pennylane_qiskit/qiskit_device2.py

```diff
@@ -0,0 +1,606 @@
+                "on samples.",
+                UserWarning,
+            )
+
+            shots = 1024
+
+        # use device shots, even if shots are defined on the Options - to do this, we update the Options
+        # shots is a required field in the Options object, and Primitive based measurements *will* use it
+        if options and options.execution.shots != 4000:  # 4000 is the default value on Options
+            warnings.warn(
+                f"Setting shots via the Options is not supported on PennyLane devices. The shots {shots} "
+                f"passed to the device will be used."
+            )
+        self.options = options or Options()
+        self.options.execution.shots = shots
+        super().__init__(wires=wires, shots=shots)
+
+        self._backend = backend
+
+        # ToDo: possibly things fail if this is not a QiskitRuntimeService - confirm and decide how to handle (SC 55725)
+        self._service = backend._service
+        self._use_primitives = use_primitives
+        self._session = session
+
+        # initial kwargs are saved and referenced every time the kwargs used for transpilation and execution
+        self._init_kwargs = kwargs
+        # _kwargs are used instead of the Options for performing raw sample based measurements (using old Qiskit API)
+        # the _kwargs are a combination of information from Options and _init_kwargs
+        self._kwargs = None
+        if self.options.simulator.noise_model:
+            self.backend.set_options(noise_model=self.options.simulator.noise_model)
+
+        # Perform validation against backend
+        b = self.backend
+        if len(self.wires) > int(b.configuration().n_qubits):
+            raise ValueError(
+                f"Backend '{backend}' supports maximum {b.configuration().n_qubits} wires"
+            )
+
+        self.reset()
+        self._update_kwargs()
+
+    @property
+    def backend(self):
+        """The Qiskit backend object.
+
+        Returns:
+            qiskit.providers.backend: Qiskit backend object.
+        """
+        return self._backend
+
+    @property
+    def service(self):
+        """The QiskitRuntimeService service.
+
+        Returns:
+            qiskit.qiskit_ibm_runtime.QiskitRuntimeService
+        """
+        return self._service
+
+    @property
+    def session(self):
+        """The QiskitRuntimeService session.
+
+        Returns:
+            qiskit.qiskit_ibm_runtime.Session
+        """
+        return self._session
+
+    @property
+    def num_wires(self):
+        return len(self.wires)
+
+    def update_session(self, session):
+        self._session = session
+
+    def reset(self):
+        self._current_job = None
+
+    def stopping_condition(self, op: qml.operation.Operator) -> bool:
+        """Specifies whether or not an observable is accepted by DefaultQubit."""
+        return op.name in self.operations
```

Comment:
From what I can tell, DefaultQubitDevice uses name comparison instead of type checking because the "accepted" observables are not limited to those of type qml.operation.Observable. The observables dict in DefaultQubitDevice also contains operators like "Sum", which is not of type qml.operation.Observable. To have the same/similar functionality, the type check solution would look something like this: (might be wrong) instance(op, (qml.operation.Observable, qml.ops.op_math.composite.Compositeop)). I guess it's just easier to keep a dictionary of allowed observables/operations and compare names than to track all the classes? Anyway, I edited the docstrings but left the original code as-is.